### DISCUSSION TITLE
Size of DB backup files

### DISCUSSION THREAD

**Discussion by natrlhy** [teslamate-org/teslamate#2437] [Q&A]
I'm curious how large others db backup files are. I've been using TeslaMate since 2019. Is this size normal or is there something I should be doing to prune things safely without data loss?

`-rw-r--r--  1 root root 586423425 Feb  2 17:01 teslamate_db.bck_3_Wednesday_Feb`

I'm just doing a standard `docker-compose exec -T db pg_dump`

**Comment by cwanja** [comment]
I know when I backed up my last database before selling my 2019 Model S, the TeslaMate backup was around 250MB. I am pretty sure `586423425` is in bytes. Convert that to MB and you get 586MB, or just over half a GB. Seems accurate. Depends on the amount of driving.

If you want to see the `ls` output in a more readable format, consider using this `-h` flag in your command. This will make the output human readable.

**Comment by natrlhy** [comment]
Has anyone used compression of their `pg_dump` data? In reading the PostgresSQL documentation, one would have to use `pg_restore` with a compressed `pg_dump -Fc` option.

I did some timing tests, and feel compressing the backup with `pg_dump`'s native compression is beneficial, but I don't want to complicate the restore process:

Simple dump:

```
root@teslamate:~/docker# time docker-compose exec -T db pg_dump -U teslamate teslamate > teslamate_db.bck_`date +%u_%A_%b`
real    3m16.448s
-rw-r--r--  1 root root 562M Feb  4 17:34 teslamate_db.bck_5_Friday_Feb
```

Native `pg_dump` standard compression:

```
root@teslamate:~/docker# time docker-compose exec -T db pg_dump -Fc -U teslamate teslamate > teslamate_db.bck_`date +%u_%A_%b`.gz
real    2m38.746s
-rw-r--r--  1 root root  72M Feb  4 17:45 teslamate_db.bck_5_Friday_Feb.gz
```
`gzip` dump (takes longer)

```
root@teslamate:~/docker# time docker-compose exec -T db pg_dump -U teslamate teslamate | gzip -9 > teslamate_db.bck_`date +%u_%A_%b`.gz
real    5m36.952s
-rw-r--r--  1 root root  72M Feb  4 17:51 teslamate_db.bck_5_Friday_Feb.gz
```

**Reply by cwanja** [reply]
Never experimented with it. But were you able to restore the compressed dump without error? If so, and you are concerned with space, I’d go that route.

**Reply by natrlhy** [reply]
I've been testing on a WSL2 instance on my WIndows PC but I don't trust whats going on :)

I've tried `docker-compose exec -T database pg_restore -U teslamate -d teslamate < teslamate_db_02_04_2022.bck.gz` and got no errors (also got no output with it) and it restored all my Geo-Fences, the DB size looks to be correct, but when I bring up Grafana...

![image](https://user-images.githubusercontent.com/19696424/152630984-9b6553f1-5e5f-49c7-8972-9b14bf7fea87.png)

I get the same regardless of restoring the compressed dump or the simple (plain text) dump on my WSL2 CLI...

**Reply by cwanja** [reply]
What do the database or Grafana logs show? I assume you did the drop the database and reinitialize step here? https://docs.teslamate.org/docs/maintenance/backup_restore/#restore

**Reply by bahree** [reply]
I run a cron job as a daily backup and I export it, and then zip up the file - that has worked out pretty well. Might be worth exploring is possible.

**Comment by CrowMash** [comment]
I've been using TeslaMate for 5 months on one car (7 525 km recorded).

`pg_dump` creates a 190Mo file.
7zip creates a 19Mo file.

I set up a scheduled task on Windows that calls this .sh script: 
```
#!/bin/bash

date=`date +%F_%H-%M`
cd "D:\TeslaMate"
docker-compose exec -T database pg_dump -U teslamate teslamate > D:/TeslaMate/Backups_DB/"$date"_teslamate.bck
7z a D:/TeslaMate/Backups_DB_ZIP/"$date"_teslamate.zip D:/TeslaMate/Backups_DB/"$date"_teslamate.bck
```

Then, the zipped file is sent to a cloud save.

Finally, if I need to restore the DB, I manually download, unzip and import my backup.
